tests/unit/amo/reducers/testCarousels.js

```diff
@@ -0,0 +1,86 @@
+import knuthShuffle from 'knuth-shuffle';
+
+import carouselsReducer, {
+  initialState,
+  setCarouselOrder,
+} from 'core/reducers/carousels';
+
+
+describe(__filename, () => {
+  const defaultParams = { name: 'MegaCarousel', sections: ['1', '2', '3'] };
+
+  it('defaults to empty', () => {
+    const state = carouselsReducer(initialState, {});
+
+    expect(state).toEqual({});
+  });
+
+  it('creates a new key if one is not found', () => {
+    const knuthShuffleSpy = sinon.spy(knuthShuffle, 'knuthShuffle');
+    const state = carouselsReducer(initialState, setCarouselOrder({
+      name: 'NotAKnownKey1234',
+      sections: [1, 2, 3],
+    }));
+
+    sinon.assert.notCalled(knuthShuffleSpy);
+    expect(state).toMatchObject({
+      NotAKnownKey1234: { order: [0, 1, 2] },
+    });
+  });
+
+  it('randomizes the order if random prop is passed', () => {
```

Comment:
It might be nice to add an assertion to make sure the original data isn't modified in terms of order by the shuffle.

Reply:
Oh, sure thing.